src/PaddleEnum.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Paddle\SDK;
+
+use MyCLabs\Enum\Enum;
+
+class PaddleEnum extends Enum
+{
+    protected const Undefined = 'undefined';
+
+    public static function from($value): static
+    {
+        try {
+            return parent::from($value);
+        } catch (\UnexpectedValueException $e) {
+            $enum = self::Undefined();
+            $enum->value = $value;
+
+            return $enum;
+        }
+    }
+
+    private static function Undefined(): self
+    {
+        return new static(self::Undefined);
+    }
```

Comment:
thinking about this again, is this actually required? I've not tested but I'd assume by default it would result in the same thing 🤔

Reply:
✅ Not needed, dropped